docs/snippets/ov_dynamic_shapes.c

```diff
@@ -0,0 +1,254 @@
+//
+#include <openvino/c/openvino.h>
+
+void reshape_with_dynamics() {
+{
+//! [ov_dynamic_shapes:reshape_undefined]
+ov_core_t* core = NULL;
+ov_core_create(&core);
+
+ov_model_t* model = NULL;
+ov_core_read_model(core, "model.xml", NULL, &model);
+
+// Set one static dimension (= 1) and another dynamic dimension (= Dimension())
+{
+ov_partial_shape_t partial_shape;
+int64_t rank = 2;
+ov_dimension_t dims[2] = {{1, 1}, {-1, -1}};
+ov_partial_shape_create(rank, dims, &partial_shape);
+ov_model_reshape(model, "tensor_name", &partial_shape, 1); // {1,?}
+}
+
+// Or set both dimensions as dynamic if both are going to be changed dynamically
+{
+ov_partial_shape_t partial_shape;
+int64_t rank = 2;
+ov_dimension_t dims[2] = {{-1, -1}, {-1, -1}};
+ov_partial_shape_create(rank, dims, &partial_shape);
+ov_model_reshape(model, "tensor_name", &partial_shape, 1); // {?,?}
+}
+//! [ov_dynamic_shapes:reshape_undefined]
+
+//! [ov_dynamic_shapes:reshape_bounds]
+// Both dimensions are dynamic, first has a size within 1..10 and the second has a size within 8..512
+{
+ov_partial_shape_t partial_shape;
+int64_t rank = 2;
+ov_dimension_t dims[2] = {{1, 10}, {8, 512}};
+ov_partial_shape_create(rank, dims, &partial_shape);
+ov_model_reshape(model, "tensor_name", &partial_shape, 1); // {1..10,8..512}
+}
+
+// Both dimensions are dynamic, first doesn't have bounds, the second is in the range of 8..512
+{
+ov_partial_shape_t partial_shape;
+int64_t rank = 2;
+ov_dimension_t dims[2] = {{-1, -1}, {8, 512}};
+ov_partial_shape_create(rank, dims, &partial_shape);
+ov_model_reshape(model, "tensor_name", &partial_shape, 1); // {?,8..512}
+}
+//! [ov_dynamic_shapes:reshape_bounds]
+}
+
+{
+ov_core_t* core = NULL;
+ov_core_create(&core);
+ov_model_t* model = NULL;
+ov_core_read_model(core, "model.xml", NULL, &model);
+//! [ov_dynamic_shapes:print_dynamic]
+// Print output partial shape
+ov_output_port_t* output_port = NULL;
+ov_model_output(model, &output_port);
+ov_partial_shape_t partial_shape;
+ov_port_get_partial_shape(output_port, &partial_shape);
+char * str_partial_shape = ov_partial_shape_to_string(partial_shape);
+printf("The output partial shape: %s", str_partial_shape);
+
+// Print input partial shape
+ov_output_port_t* input_port = NULL;
+ov_model_input(model, &input_port);
+ov_partial_shape_t partial_shape;
+ov_port_get_partial_shape(input_port, &partial_shape);
+char * str_partial_shape = ov_partial_shape_to_string(partial_shape);
+printf("The input partial shape: %s", str_partial_shape);
+//! [ov_dynamic_shapes:print_dynamic]
+}
+
+{
+ov_core_t* core = NULL;
+ov_core_create(&core);
+
+//! [ov_dynamic_shapes:detect_dynamic]
+ov_model_t* model = NULL;
+ov_core_read_model(core, "model.xml", NULL, &model);
+
+ov_output_port_t* input_port = NULL;
+ov_model_input_by_index(model, 0, &input_port);
+ov_partial_shape_t partial_shape;
+ov_port_get_partial_shape(input_port, &partial_shape);
+if (ov_partial_shape_is_dynamic(partial_shape)) {
```

Comment:
It should follow C++ snippet code for dynamic shape. 
Will be possible add below api to check whether model is dynamic?
`bool ov_model_is_dynamic(const ov_model_t* model)`

Reply:
In here, C++ check dynamic or not by `model->output(0).get_partial_shape().is_dynamic()`, which according to the partial shape. So using `ov_partial_shape_is_dynamic()` for C